core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/JsonRpcMessageReceiver.java

```diff
@@ -91,18 +91,36 @@ private void processResponse(String endpointId, String innerMessage) {
     responseDispatcher.dispatch(endpointId, response);
   }
 
-  private void processRequest(String endpointId, String innerMessage) {
-    JsonRpcRequest request = null;
-    try {
-      request = jsonRpcUnmarshaller.unmarshalRequest(innerMessage);
-      requestDispatcher.dispatch(endpointId, request);
-    } catch (JsonRpcException e) {
-      if (request == null || request.getId() == null) {
-        errorTransmitter.transmit(endpointId, e);
-      } else {
-        errorTransmitter.transmit(
-            endpointId, new JsonRpcException(e.getCode(), e.getMessage(), request.getId()));
+  private class JsonRpcRunnable implements Runnable {
```

Comment:
I think it would be better to choose a name which shows the responsibility of this class, like `ProcessRequestTask`. WDYT?